### PR TITLE
bug fix , configuration format lost when rollback

### PR DESCRIPTION
**Pull request by HollisGW**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

nacos-config

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [x] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

File: config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java [hunk 1]
```diff
@@ -508,7 +509,8 @@ public Boolean doInTransaction(TransactionStatus status) {
                     String configTags = configAdvanceInfo == null ? null : (String) configAdvanceInfo.get("config_tags");
                     addConfiTagsRelationAtomic(configId, configTags, configInfo.getDataId(), configInfo.getGroup(),
                         configInfo.getTenant());
-                    insertConfigHistoryAtomic(0, configInfo, srcIp, srcUser, time, "I");
+                    String type = configAdvanceInfo == null ? null : (String) configAdvanceInfo.get("type");
```
Comment (chuntaojun) [inline config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java]: I think if type not set, The text can be set to the default value of type

Comment (HollisGW) [reply config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java]: Good advice. I'll change it.

File: config/src/main/resources/META-INF/schema.sql [hunk 2]
```diff
@@ -38,6 +38,7 @@ CREATE TABLE his_config_info (
   src_user varchar(128),
   src_ip varchar(20) DEFAULT NULL,
   op_type char(10) DEFAULT NULL,
+  type varchar(64) DEFAULT NULL,
```
Comment (chuntaojun) [inline config/src/main/resources/META-INF/schema.sql]: Mysql script also need to change

Comment (HollisGW) [reply config/src/main/resources/META-INF/schema.sql]: ok

File: config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java [hunk 3]
```diff
@@ -599,7 +601,8 @@ public Boolean doInTransaction(TransactionStatus status) {
                         addConfiTagsRelationAtomic(oldConfigInfo.getId(), configTags, configInfo.getDataId(),
                             configInfo.getGroup(), configInfo.getTenant());
                     }
-                    insertConfigHistoryAtomic(oldConfigInfo.getId(), oldConfigInfo, srcIp, srcUser, time, "U");
+                    String type = configAdvanceInfo == null ? null : (String) configAdvanceInfo.get("type");
```
[the remark this answers is not in the document]
Comment (HollisGW) [reply config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java]: 默认type为text不是在这里改的，在insertConfigHistoryAtomic这个方法里统一处理的

[the remark this answers is not in the document]
Comment (chuntaojun) [reply config/src/main/java/com/alibaba/nacos/config/server/service/PersistService.java]: 哦...，我的失误...

**Comment by nkorange** [comment]
Please add some test cases.

**Comment by HollisGW** [comment]
> Please add some test cases.

Already added

**Comment by KomachiSion** [comment]
Please fix conflict

**Comment by KomachiSion** [comment]
There are conflicts and author has no response for a long time. I will close issue, if you want to contribute again. Please open a new PR.

**Pull request closed without merging**